### PR TITLE
Set up travis-ci and flake8

### LINKED ISSUE #4: Use flake8 linting; hook up to CI.

### PR DESCRIPTION
**Pull request by pdehaan**
Fixes #4

### CODE REVIEW

**Comment by pdehaan** [comment]
Yay! Travis works, and fails with the following:
```
➜  universal-search-recommendation git:(master) ✗ flake8 .
./app/search/recommendation.py:1:1: F401 'memcached' imported but unused
./app/search/classification/domain.py:1:1: F401 'urlparse' imported but unused
./app/search/classification/wikipedia.py:1:1: F401 'urlparse' imported but unused
./app/search/classification/wikipedia.py:6:1: F401 'memcached' imported but unused
./app/search/query/base.py:4:1: F401 'memcached' imported but unused
./app/search/query/yahoo.py:8:1: F401 'memorize' imported but unused
./app/search/query/yahoo.py:9:1: F401 'memcached' imported but unused
./app/search/suggest/base.py:2:1: F401 'memcached' imported but unused
./app/search/suggest/bing.py:3:1: F401 'memorize' imported but unused
./app/search/suggest/bing.py:4:1: F401 'memcached' imported but unused
```

I'll remove the unused bits and repush.

**Comment by pdehaan** [comment]
Feel free to angrily close this. I just happened to be doing Flake8 in a different repo and thought I'd play in your sandbox.

File: app/main.py [hunk 1]
```diff
@@ -12,6 +12,7 @@
 sys.path.append(os.path.dirname(__file__))
 app = Flask(__name__)
 
+
```
Comment (pdehaan) [inline app/main.py]: flake8 says:
```
./app/main.py:15:1: E302 expected 2 blank lines, found 1
```

File: app/search/query/yahoo.py [hunk 2]
```diff
@@ -24,7 +22,7 @@ def fetch(self, query):
             'oauth_nonce': oauth2.generate_nonce(),
             'oauth_timestamp': str(int(time.time())),
             'oauth_version': '1.0',
-            'q' : quote_plus(query)
+            'q': quote_plus(query)
```
Comment (pdehaan) [inline app/search/query/yahoo.py]: flake8 says:
```
./app/search/query/yahoo.py:27:16: E203 whitespace before ':'
```

**Comment by chuckharmston** [comment]
Nifty! Mind submitting this as a pull request against the branch `2-unit-tests` on chuckharmston/universal-search-recommendation, rather than against `master` on mozilla/universal-search-recommendation? That will be a slightly long-running branch that covers much of the same ground.

**Comment by pdehaan** [comment]
@chuckharmston Submitted as https://github.com/chuckharmston/universal-search-recommendation/pull/1

I'll close this unmerged PR out.

**Pull request closed without merging**